cmd/mysql-state-monitor/main.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"strings"
+)
+
+type MySQLState string
+
+const (
+	MySQLReady   MySQLState = "ready"
+	MySQLDown    MySQLState = "down"
+	MySQLStartup MySQLState = "startup"
+	MySQLUnknown MySQLState = "unknown"
+)
+
+func parseDatum(datum string) MySQLState {
+	lines := strings.Split(datum, "\n")
+
+	if lines[0] == "READY=1" {
+		return MySQLReady
+	}
+
+	if lines[0] == "STOPPING=1" {
+		return MySQLDown
+	}
+
+	if strings.HasPrefix(lines[0], "STATUS=") {
+		status := strings.TrimPrefix(lines[0], "STATUS=")
+		switch status {
+		case "Server is operational":
+			return MySQLReady
+		case "Server shutdown in progress":
+			return MySQLDown
+		case "Server startup in progress",
+			"Data Dictionary upgrade in progress",
+			"Data Dictionary upgrade complete",
+			"Server upgrade in progress",
+			"Server upgrade complete",
+			"Server downgrade in progress",
+			"Server downgrade complete",
+			"Data Dictionary upgrade from MySQL 5.7 in progress",
+			"Data Dictionary upgrade from MySQL 5.7 complete",
+			"Server shutdown complete": // we treat this as startup because during init, MySQL notifies this even if it's up
+			return MySQLStartup
+		}
+	}
+
+	return MySQLUnknown
+}
+
+func main() {
```

Comment:
Can we have some e2e test, like reading this file and checking if its ok?
Maybe a case in init-deploy tests?